src/edit_mode/vi/mod.rs

```diff
@@ -143,7 +142,16 @@ impl EditMode for Vi {
                 (_, KeyModifiers::NONE, KeyCode::Esc) => {
                     self.cache.clear();
                     self.mode = ViMode::Normal;
-                    ReedlineEvent::Multiple(vec![ReedlineEvent::Esc, ReedlineEvent::Repaint])
+                    ReedlineEvent::Multiple(vec![
+                        // Move left then right to clear the selection.
+                        // Order matters here: this makes sure cursor does not move at end of line
+                        ReedlineEvent::Edit(vec![
+                            EditCommand::MoveLeft { select: false },
+                            EditCommand::MoveRight { select: false },
+                        ]),
```

Comment:
As we are controlling things anyways, wouldn't a new `ReedlineEvent::ResetSelection` be easier and much clearer?